# Day 7 - Pandas Dataset Summary Report

In this notebook, I am using Pandas to load and understand a student dataset. I will check the rows, columns, data types, basic statistics and missing values.

In [1]:
import pandas as pd

## 1. Load the CSV File


In [3]:
df = pd.read_csv('/content/Day7_Student_Dataset.csv')
df

,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
0,1001,Aarav Sharma,19,Data Science,87.0,91.0,Srinagar
1,1002,Zoya Khan,20,AI,92.0,95.0,Delhi
2,1003,Rohan Mehta,18,Computer Science,76.0,82.0,Mumbai
3,1004,Ananya Singh,21,Data Science,81.0,88.0,Srinagar
4,1005,Kabir Ali,19,AI,69.0,76.0,Lucknow
5,1006,Ishita Gupta,20,Computer Science,88.0,93.0,Bengaluru
6,1007,Aditya Verma,22,Data Science,95.0,97.0,Pune
7,1008,Sara Ahmed,19,AI,73.0,NaN,Hyderabad
8,1009,Arjun Nair,18,Computer Science,84.0,86.0,Kochi
9,1010,Mehak Bhat,21,Data Science,79.0,90.0,Srinagar


## 2. First 5 Rows

`head()` shows the first 5 rows of the dataset.

In [4]:
df.head()

,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
0,1001,Aarav Sharma,19,Data Science,87.0,91.0,Srinagar
1,1002,Zoya Khan,20,AI,92.0,95.0,Delhi
2,1003,Rohan Mehta,18,Computer Science,76.0,82.0,Mumbai
3,1004,Ananya Singh,21,Data Science,81.0,88.0,Srinagar
4,1005,Kabir Ali,19,AI,69.0,76.0,Lucknow


## 3. Last 5 Rows

`tail()` shows the last 5 rows.

In [5]:
df.tail()

,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
20,1021,Aman Kumar,18,Computer Science,74.0,78.0,Delhi
21,1022,Alina Shah,19,Data Science,96.0,98.0,Srinagar
22,1023,Nikhil Sood,22,AI,78.0,83.0,Pune
23,1024,Maryam Khan,20,Computer Science,85.0,88.0,Hyderabad
24,1025,Reyansh Jain,21,Data Science,80.0,84.0,Kolkata


## 4. Random Rows

`sample()` gives some random rows from the dataset.

In [6]:
df.sample(5)

,Student_ID,Name,Age,Course,Marks,Attendance_Percentage,City
20,1021,Aman Kumar,18,Computer Science,74.0,78.0,Delhi
1,1002,Zoya Khan,20,AI,92.0,95.0,Delhi
15,1016,Hiba Sheikh,21,Data Science,94.0,96.0,Srinagar
12,1013,Rahul Das,23,Data Science,72.0,81.0,Kolkata
17,1018,Manya Rao,22,Computer Science,70.0,74.0,Bengaluru


## 5. Shape

`shape` tells us the number of rows and columns.

In [7]:
df.shape

(25, 7)

In [8]:
print('Number of rows:', df.shape[0])
print('Number of columns:', df.shape[1])

Number of rows: 25
Number of columns: 7


## 6. Columns

This shows all the column names present in the dataset.

In [9]:
df.columns

Index(['Student_ID', 'Name', 'Age', 'Course', 'Marks', 'Attendance_Percentage',
       'City'],
      dtype='object')

## 7. Index

The index is used to identify the rows in the DataFrame.

In [10]:
df.index

RangeIndex(start=0, stop=25, step=1)

## 8. Data Types

Using `dtypes`, I can see the type of data stored in each column.

In [11]:
df.dtypes

,0
Student_ID,int64
Name,object
Age,int64
Course,object
Marks,float64
Attendance_Percentage,float64
City,object


## 9. Basic Information

`info()` gives information about the DataFrame, including columns, non-null values and data types.

In [12]:
df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 25 entries, 0 to 24
Data columns (total 7 columns):
 #   Column                 Non-Null Count  Dtype  
---  ------                 --------------  -----  
 0   Student_ID             25 non-null     int64  
 1   Name                   25 non-null     object 
 2   Age                    25 non-null     int64  
 3   Course                 25 non-null     object 
 4   Marks                  24 non-null     float64
 5   Attendance_Percentage  24 non-null     float64
 6   City                   24 non-null     object 
dtypes: float64(2), int64(2), object(3)
memory usage: 1.5+ KB


## 10. Descriptive Statistics

`describe()` gives some basic statistical values for the numerical columns.

In [13]:
df.describe()

,Student_ID,Age,Marks,Attendance_Percentage
count,25.000000,25.000000,24.000000,24.000000
mean,1013.000000,20.000000,82.250000,87.083333
std,7.359801,1.414214,8.689074,7.252686
min,1001.000000,18.000000,68.000000,72.000000
25%,1007.000000,19.000000,75.500000,82.750000
50%,1013.000000,20.000000,82.500000,88.000000
75%,1019.000000,21.000000,89.250000,92.250000
max,1025.000000,23.000000,96.000000,98.000000


## 11. Checking Missing Values

I used `isnull()` to check where values are missing and `sum()` to count them.

In [14]:
df.isnull().sum()

,0
Student_ID,0
Name,0
Age,0
Course,0
Marks,1
Attendance_Percentage,1
City,1


## 13. Basic Observations

- There are **25 rows and 7 columns** in the dataset.
- The columns are `Student_ID`, `Name`, `Age`, `Course`, `Marks`, `Attendance_Percentage` and `City`.
- `Student_ID` and `Age` are integer (`int64`) columns.
- `Marks` and `Attendance_Percentage` are floating-point (`float64`) columns.
- `Name`, `Course` and `City` are stored as `object` columns.
- There is **1 missing value** in `Marks`, **1 missing value** in `Attendance_Percentage`, and **1 missing value** in `City`.
- The other columns do not have missing values.
- From `describe()`, the average age is **20 years**, the average marks are **82.25**, and the average attendance is about **87.08%**.

## Conclusion

By using Pandas, I was able to get a quick idea about the student dataset. I checked its shape, columns, index and data types, and also used `head()`, `tail()`, `sample()`, `info()`, `describe()` and `isnull().sum()`. The dataset is small and has only a few missing values, so it is easy to inspect and work with.